ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/toolAdmin/view/subviews/ToolDeviceListEditor.java

```diff
@@ -0,0 +1,139 @@
+import com.sencha.gxt.data.shared.ListStore;
+import com.sencha.gxt.data.shared.ModelKeyProvider;
+import com.sencha.gxt.widget.core.client.Composite;
+import com.sencha.gxt.widget.core.client.event.CancelEditEvent;
+import com.sencha.gxt.widget.core.client.form.TextField;
+import com.sencha.gxt.widget.core.client.grid.ColumnConfig;
+import com.sencha.gxt.widget.core.client.grid.ColumnModel;
+import com.sencha.gxt.widget.core.client.grid.Grid;
+import com.sencha.gxt.widget.core.client.grid.editing.AbstractGridEditing;
+import com.sencha.gxt.widget.core.client.grid.editing.ClicksToEdit;
+import com.sencha.gxt.widget.core.client.grid.editing.GridEditing;
+import com.sencha.gxt.widget.core.client.grid.editing.GridRowEditing;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class ToolDeviceListEditor extends Composite implements IsEditor<Editor<List<ToolDevice>>> {
+
+    private GridEditing<ToolDevice> editing;
+    private Grid<ToolDevice> grid;
+    private static final String TOOL_DEVICE_MODEL_KEY = "model_key";
+    private int unique_device_id;
+    private ListStoreEditor<ToolDevice> listStoreEditor;
+
+    @UiField ListStore<ToolDevice> listStore;
+    @UiField(provided = true) ToolAdminView.ToolAdminViewAppearance appearance;
+
+    @Inject
+    public ToolDeviceListEditor(ToolDeviceProperties toolDeviceProperties,
+                                ToolAdminView.ToolAdminViewAppearance appearance) {
+        this.appearance = appearance;
+        listStore = new ListStore<>(new ModelKeyProvider<ToolDevice>() {
+            @Override
+            public String getKey(ToolDevice item) {
+                return getDeviceTag(item);
+            }
+        });
+        listStoreEditor = new ListStoreEditor<>(listStore);
+        listStore.setAutoCommit(true);
+
+        ColumnConfig<ToolDevice, String> hostPath = new ColumnConfig<>(toolDeviceProperties.hostPath(),
+                                                                       appearance.containerDevicesHostPathWidth(),
+                                                                       appearance.containerDevicesHostPathLabel());
+        ColumnConfig<ToolDevice, String> containerPath = new ColumnConfig<>(toolDeviceProperties.containerPath(),
+                                                                            appearance.containerDevicesContainerPathWidth(),
+                                                                            appearance.containerDevicesContainerPathLabel());
+
+        List<ColumnConfig<ToolDevice, ?>> columns = new ArrayList<>();
+        columns.add(hostPath);
+        columns.add(containerPath);
+        ColumnModel<ToolDevice> cm = new ColumnModel<>(columns);
+
+        grid = new Grid<>(listStore, cm);
+        grid.setHeight(100);
+
+        editing = new GridRowEditing<>(grid);
+        editing.addEditor(hostPath, new TextField());
+        editing.addEditor(containerPath, new TextField());
```

Comment:
Not sure how I put `setAllowBlank` in one but not the other, but should be fixed now.